src/trusted/service_runtime/nacl_syscall_common.c

```diff
@@ -941,6 +946,7 @@ int32_t NaClSysRead(struct NaClAppThread  *natp,
   /* This cast is safe because we clamped count above.*/
   retval = (int32_t) read_result;
 out:
+  NaClFastMutexUnlock(&nap->desc_mu);
```

Comment:
Hmm... so unref wasn't being called here at out, now I'm not as sure about any of this... that means the ref increment was actually persisting, but until where?